### PR TITLE
add guide for drizzle + neon

### PR DESCRIPTION
**Pull request by ananis25**
### What does this PR do?

This adds a tutorial on connecting to Neon database using Drizzle ORM from Bun. 

(I write technical content for Neon)

- [x] Documentation or TypeScript types (it's okay to leave the rest blank in this case)

### CODE REVIEW

File: docs/guides/ecosystem/neon-drizzle.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+---
+name: Use Neon Postgres using Drizzle ORM with Bun
```
Comment (Jarred-Sumner) [inline docs/guides/ecosystem/neon-drizzle.md]: `with Bun` is automatically appended to the names

```suggestion
name: Use Neon Postgres using Drizzle ORM
```

File: docs/guides/ecosystem/neon-drizzle.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+---
+
+[Neon](https://neon.tech/) is a fully managed serverless Postgres. Neon separates compute and storage to offer modern developer features such as autoscaling, branching, bottomless storage, and more.
```
Comment (Jarred-Sumner) [inline docs/guides/ecosystem/neon-drizzle.md]: Can we combine this and the below "getting started" section into one section? That way they see code to copy paste immediately

**Comment by ananis25** [comment]
Thank you, edited the post.

**Comment by evanshortiss** [comment]
@ananis25 thank you for making this guide, from Neon ❤️

**Pull request merged**